src/main/java/it/gov/pagopa/payhub/activities/service/debtposition/custom/fine/DebtPositionFineProcessor.java

```diff
@@ -0,0 +1,54 @@
+package it.gov.pagopa.payhub.activities.service.debtposition.custom.fine;
+
+import it.gov.pagopa.payhub.activities.dto.debtposition.HandleFineDebtPositionResult;
+import it.gov.pagopa.payhub.activities.dto.debtposition.syncwfconfig.FineWfExecutionConfig;
+import it.gov.pagopa.pu.debtposition.dto.generated.*;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Objects;
+
+@Lazy
+@Service
+public class DebtPositionFineProcessor {
+
+    private final DebtPositionFineNotificationDateProcessor notificationDateProcessor;
+
+    public DebtPositionFineProcessor(DebtPositionFineNotificationDateProcessor notificationDateProcessor) {
+        this.notificationDateProcessor = notificationDateProcessor;
+    }
+
+    public HandleFineDebtPositionResult processFine(DebtPositionDTO debtPositionDTO, FineWfExecutionConfig executionParams) {
+        HandleFineDebtPositionResult handleFineDebtPositionResult = notificationDateProcessor.processNotificationDate(debtPositionDTO, executionParams);
+
+        if (handleFineDebtPositionResult.getReductionEndDate() == null || handleFineDebtPositionResult.getReductionEndDate().isAfter(OffsetDateTime.now())) {
```

Comment:
let's add a comment to let be more clearer what's happening
```suggestion

        // If we are currently on the reduction period 
        if (handleFineDebtPositionResult.getReductionEndDate() == null || handleFineDebtPositionResult.getReductionEndDate().isAfter(OffsetDateTime.now())) {
```